DAL/user.DAL.js

```diff
@@ -0,0 +1,42 @@
+// get all employees -- need to move ot DAL
+getAllUsers = () => {
+    return {1 : "Gilad", 2 : "Adva", 3: "Ofir", 4: "Dave"};
+};
+
+const getUserById = (id) => {
+    if(id == null)
+        return 'No valid id';
+    let users = getAllUsers();
+    if (id in users) {
+        return users[id];
+    }
+    return ["This id doesn't exist"];
+    };
+
+const setHomeAddress = (id, home_address) => {
+    if(id == null)
+        return 'Missing id';
+    let users = getAllUsers();
+    if (id in users) {
+        return [`The home address of ${users[id]} changed to ${home_address}`];
+    }
+    return ['Invalid id']
+};
+
+// add check if there is active request
+const setVacationRequest = (id, startDate, finishDate) => {
+    if(id == null)
+        return 'Missing id';
+    let users = getAllUsers();
+    if (id in users) {
+        return [`${users[id]} sent vacation request`];
+    }
+    return ['Invalid id']
+};
```

Comment:
vacation request belongs to Ofir, you don't need this functionality in your users.dal. It will be under vacation.DAL